File: CPP/Day15.cpp
/*

Day 15: Program to check whether the given number is palindrome or not

*/

//Program
#include <iostream>
using namespace std;

bool isPalindrome(int num) {
    if (num < 0) {
        return false;
    }
    int rev = 0;
    int original = num;
    while (original > 0) {
        rev = rev * 10 + original % 10;
        original /= 10;
    }
    return num == rev;
}

int main() {
    int num;
    cout<<"Enter the number: ";
    cin >> num;

    if (isPalindrome(num)) {
        cout << num << " is PALINDROME." << endl;
    } 
    else {
        cout << num << " is NOT PALINDROME." << endl;
    }
    return 0;
}




/*


Input & Output:

Enter the number: 123454321
123454321 is PALINDROME.


*/

File: Python/Day15.py
"""

Day 15:	Program to check whether the given number is palindrome or not.


"""

#Program
def palindrome(m):
    rev = 0
    n = m
    
    while n > 0:
        rev = (rev * 10) + (n % 10)
        n = n // 10
    
    if (m == rev):
        print(f"{m} is a PALINDROME.")
    else:
        print(f"{m} is NOT PALINDROME")
        

n = int(input("Enter the number: "))
palindrome(n)





"""

Input and Output: 

Enter the number: 4334
4334 is a PALINDROME.

"""